MapboxNavigation/NavigationMapView.swift

```diff
@@ -234,6 +234,7 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     
     deinit {
         suspendNotifications()
+        batteryStateObservation = nil
```

Comment:
Do we really need this observer? It seems that the logic being done on Line [229](https://github.com/mapbox/mapbox-navigation-ios/blob/86b69b45a4519679abfda1e642cdf2fa53e5be86/MapboxNavigation/NavigationMapView.swift#L229) could be handled via a computed property.

```swift
var isPluggedIn: Bool { return [.charging, .full].contains(UIDevice.current.batteryState) }
```

Reply:
In fact, this could be an extension on `UIDevice`. 

```swift
UIDevice.current.isPluggedIn
```

Reply:
We need to observe it to trigger on change of charging state and modify the frame rate accordingly.

Reply:
@frederoni I hear you, but the local `batteryState` property wasn't acted upon on-change (like with a `didSet` or something), the KVO was just updating the local property. Thus you can replace the bound local property with a computed one.